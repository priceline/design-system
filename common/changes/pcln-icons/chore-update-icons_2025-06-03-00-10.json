{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "Update to styled-system v5 on pcln-icons, move to dev dependency and peer dependency. Use Node v20 for the build process.",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}