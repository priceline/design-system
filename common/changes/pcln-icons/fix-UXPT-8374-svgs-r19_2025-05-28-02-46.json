{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "replace defaultProps usage in svgr template file",
      "type": "patch"
    }
  ],
  "packageName": "pcln-icons"
}