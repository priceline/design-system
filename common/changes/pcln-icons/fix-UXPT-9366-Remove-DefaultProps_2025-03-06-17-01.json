{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "remove defaultProps",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}