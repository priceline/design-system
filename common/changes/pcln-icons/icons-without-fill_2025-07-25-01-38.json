{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "UXPT-10383: rounded Material SVGs from filled to unfilled",
      "type": "major"
    }
  ],
  "packageName": "pcln-icons"
}