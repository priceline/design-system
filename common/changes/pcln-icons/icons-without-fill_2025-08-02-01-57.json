{
  "changes": [
    {
      "comment": "allow styled-components 6 as a peer",
      "type": "patch",
      "packageName": "pcln-icons"
    }
  ],
  "packageName": "pcln-icons",
  "email": "steven.dalonzo@priceline.com"
}