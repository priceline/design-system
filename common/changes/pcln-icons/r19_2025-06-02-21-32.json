{
  "changes": [
    {
      "comment": "React 19 ecosystem upgrade",
      "type": "major",
      "packageName": "pcln-icons"
    }
  ],
  "packageName": "pcln-icons",
  "email": "steven.dalonzo@priceline.com"
}