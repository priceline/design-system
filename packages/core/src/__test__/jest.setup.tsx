import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
