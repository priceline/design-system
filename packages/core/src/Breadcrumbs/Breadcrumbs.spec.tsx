import { Flights, Home, Seat } from 'pcln-icons'
import { render } from '../__test__/testing-library'
import { Breadcrumbs } from './Breadcrumbs'

describe('Breadcrumbs', () => {
  test('renders basic', () => {
    const { asFragment } = render(
      <Breadcrumbs>
        <Breadcrumbs.Link href='https://www.priceline.com' label='Home' />
        <Breadcrumbs.Link href='https://www.priceline.com/flights/' label='Flights' />
        <Breadcrumbs.Link href='https://www.priceline.com/flights/' label='Seat Selection' />
      </Breadcrumbs>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with icons', () => {
    const { asFragment } = render(
      <Breadcrumbs>
        <Breadcrumbs.Link
          href='https://www.priceline.com'
          label='Home'
          icon={<Home color='text.light' size={16} mr={2} />}
        />
        <Breadcrumbs.Link
          href='https://www.priceline.com/flights/'
          label='Flights'
          icon={<Flights color='text.light' size={16} mr={2} />}
        />
        <Breadcrumbs.Link
          href='https://www.priceline.com/flights/'
          label='Seat Selection'
          icon={<Seat color='text.dark' size={16} mr={2} />}
        />
      </Breadcrumbs>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
