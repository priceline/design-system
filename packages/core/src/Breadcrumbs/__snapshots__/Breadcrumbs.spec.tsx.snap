// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders basic 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #4f6f8f;
}

.c2:hover {
  color: #4f6f8f;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c4 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #001023;
}

.c4:hover {
  color: #001023;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  color: #4f6f8f;
  margin: 8px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <a
      className="c2"
      color="text.light"
      href="https://www.priceline.com"
      rel={null}
      size="medium"
    >
      Home
    </a>
    <div
      className="c3"
      color="text.light"
    >
      /
    </div>
  </div>
  <div
    className="c1"
  >
    <a
      className="c2"
      color="text.light"
      href="https://www.priceline.com/flights/"
      rel={null}
      size="medium"
    >
      Flights
    </a>
    <div
      className="c3"
      color="text.light"
    >
      /
    </div>
  </div>
  <div
    className="c1"
  >
    <a
      className="c4"
      color="text.dark"
      href="https://www.priceline.com/flights/"
      rel={null}
      size="medium"
    >
      Seat Selection
    </a>
  </div>
</div>
`;

exports[`Breadcrumbs renders with icons 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  color: #4f6f8f;
  width: 16px;
  margin-right: 8px;
}

.c7 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  color: #001023;
  width: 16px;
  margin-right: 8px;
}

.c6 {
  outline: none;
}

.c3 {
  outline: none;
}

.c8 {
  outline: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #4f6f8f;
}

.c4:hover {
  color: #4f6f8f;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #001023;
}

.c9:hover {
  color: #001023;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  color: #4f6f8f;
  margin: 8px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      color="text.light"
      fill="currentcolor"
      focusable={false}
      height={16}
      role="img"
      tabIndex={-1}
      viewBox="0 -960 960 960"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M240-200h120v-200q0-17 11.5-28.5T400-440h160q17 0 28.5 11.5T600-400v200h120v-360L480-740 240-560v360zm-80 0v-360q0-19 8.5-36t23.5-28l240-180q21-16 48-16t48 16l240 180q15 11 23.5 28t8.5 36v360q0 33-23.5 56.5T720-120H560q-17 0-28.5-11.5T520-160v-200h-80v200q0 17-11.5 28.5T400-120H240q-33 0-56.5-23.5T160-200zm320-270z"
      />
    </svg>
    <a
      className="c4"
      color="text.light"
      href="https://www.priceline.com"
      rel={null}
      size="medium"
    >
      Home
    </a>
    <div
      className="c5"
      color="text.light"
    >
      /
    </div>
  </div>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c6"
      color="text.light"
      fill="currentcolor"
      focusable={false}
      height={16}
      role="img"
      tabIndex={-1}
      viewBox="0 -960 960 960"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M480-120L377-91q-14 4-25.5-4.5T340-118q0-12 3-19.5t8-11.5l69-51v-220l-291 86q-19 5-34-6t-15-31q0-15 5-25t14-15l321-189v-220q0-25 17.5-42.5T480-880q25 0 42.5 17.5T540-820v220l321 189q9 5 14 15t5 25q0 20-15 31t-34 6l-291-86v220l69 51q5 4 8 11.5t3 19.5q0 14-11.5 22.5T583-91l-103-29z"
      />
    </svg>
    <a
      className="c4"
      color="text.light"
      href="https://www.priceline.com/flights/"
      rel={null}
      size="medium"
    >
      Flights
    </a>
    <div
      className="c5"
      color="text.light"
    >
      /
    </div>
  </div>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c7 c8"
      color="text.dark"
      fill="currentcolor"
      focusable={false}
      height={16}
      role="img"
      tabIndex={-1}
      viewBox="0 -960 960 960"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M200-120q-17 0-28.5-11.5T160-160v-120q0-33 23.5-56.5T240-360h480q33 0 56.5 23.5T800-280v120q0 17-11.5 28.5T760-120q-17 0-28.5-11.5T720-160v-120H240v120q0 17-11.5 28.5T200-120zm-20-280q-25 0-42.5-17.5T120-460q0-25 17.5-42.5T180-520q25 0 42.5 17.5T240-460q0 25-17.5 42.5T180-400zm100 0v-360q0-33 23.5-56.5T360-840h240q33 0 56.5 23.5T680-760v360H280zm500 0q-25 0-42.5-17.5T720-460q0-25 17.5-42.5T780-520q25 0 42.5 17.5T840-460q0 25-17.5 42.5T780-400zm-420-80h240v-280H360v280zm0 0h240-240z"
      />
    </svg>
    <a
      className="c9"
      color="text.dark"
      href="https://www.priceline.com/flights/"
      rel={null}
      size="medium"
    >
      Seat Selection
    </a>
  </div>
</div>
`;
